parachain/pallets/ethereum2-light-client/src/lib.rs

```diff
@@ -0,0 +1,447 @@
+//! # Ethereum 2 Light Client Verifier
+//!
+//! This module implements the `Verifier` interface. Other modules should reference
+//! this module using the `Verifier` type and perform verification using `Verifier::verify`.
+#![allow(unused_variables)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+use codec::{Decode, Encode};
+use frame_support::{dispatch::DispatchResult, log, traits::Get, transactional};
+use frame_system::ensure_signed;
+use scale_info::TypeInfo;
+use sp_core::H256;
+use sp_runtime::RuntimeDebug;
+use sp_std::prelude::*;
+use std::cmp;
+
+pub use snowbridge_ethereum::Header as EthereumHeader;
+
+/// https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/sync-protocol.md#misc
+/// The minimum number of validators that needs to sign update
+const MIN_SYNC_COMMITTEE_PARTICIPANTS: u64 = 1;
+
+const EPOCHS_PER_SYNC_COMMITTEE_PERIOD: u64 = 512;
+
+const SECONDS_PER_SLOT: u64 = 12;
+
+const SLOTS_PER_EPOCH: u64 = 32;
+
+const SYNC_COMMITTEE_SIZE: u64 = 512;
+
+const UPDATE_TIMEOUT: u64 = SLOTS_PER_EPOCH * EPOCHS_PER_SYNC_COMMITTEE_PERIOD;
+
+const DOMAIN_SYNC_COMMITTEE: u64 = 1; // TODO figure out what this is
+
+type Epoch = u64;
+type Slot = u64;
+type Root = H256;
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct ValidatorIndex {
+	// TODO: Add
+}
+
+/// Beacon block header as it is stored in the runtime storage.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+// https://yeeth.github.io/BeaconChain.swift/Structs/BeaconBlockHeader.html#/s:11BeaconChain0A11BlockHeaderV9signature10Foundation4DataVvp
+// https://benjaminion.xyz/eth2-annotated-spec/phase0/beacon-chain/#beaconblock
+pub struct BeaconBlockHeader {
+	// The slot for which this block is created. Must be greater than the slot of the block defined by parentRoot.
+	pub slot: Slot,
+	// The block root of the parent block, forming a block chain.
+	pub parent_root: H256,
+	// The hash root of the post state of running the state transition through this block.
+	pub state_root: H256,
+	// BLS Signature of the block by the block proposer, TODO type isn't right yet
+	pub signature: String,
+	proposer_index: ValidatorIndex,
+}
+
+/// Sync committee as it is stored in the runtime storage.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SyncCommittee {
+	pub pubkeys: Vec<String>, // TODO most likely not a string
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SyncAggregate {
+	// 1 or 0 bit, indicates whether a sync committee participated in a vote
+	pub sync_committee_bits: Vec<u64>,
```

Comment:
Should rather use `u8` instead of `u64` for this :)

```suggestion
	pub sync_committee_bits: Vec<u8>,
```